service/lib/agama/storage/volume_conversion/to_y2storage.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Copyright (c) [2023] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage"
+require "agama/storage/volume"
+
+module Agama
+  module Storage
+    module VolumeConversion
+      # Volume conversion to Y2Storage format.
+      class ToY2Storage
+        # @param volume [Agama::Storage::Volume]
+        def initialize(volume)
+          @volume = volume
+        end
+
+        # Performs the conversion to Y2Storage format.
+        #
+        # @return [Y2Storage::VolumeSpecification]
+        def convert # rubocop:disable Metrics/AbcSize
+          Y2Storage::VolumeSpecification.new({}).tap do |target|
+            target.device = volume.device
+            target.separate_vg_name = volume.separate_vg_name
+            target.mount_point = volume.mount_path
+            target.mount_options = volume.mount_options.join(",")
+            target.proposed = true
+            target.proposed_configurable = !volume.outline.required?
+            target.fs_types = volume.outline.filesystems
+            target.fs_type = volume.fs_type if volume.fs_type
+            target.weight = 100
+            target.adjust_by_ram = volume.outline.adjust_by_ram?
+
+            sizes_conversion(target)
+            btrfs_conversion(target)
+          end
+        end
+
+      private
+
+        # @return [Agama::Storage::Volume]
+        attr_reader :volume
+
+        # @param target [Y2Storage::VolumeSpecification]
+        def sizes_conversion(target)
+          auto = volume.auto_size?
+
+          target.ignore_fallback_sizes = !auto
+          target.ignore_snapshots_sizes = !auto
+
+          target.min_size = auto ? volume.outline.base_min_size : volume.min_size
+          target.max_size = auto ? volume.outline.base_max_size : volume.max_size
```

Comment:
and here I am also a bit confused as conversion from y2storage and to y2storage looks like have different logic. So I would like to know also here rationale, why for auto it sets outline and otherwise directly on volume.

Reply:
Ok.